tests/psalm/ArrayShapeTest.php

```diff
@@ -0,0 +1,20 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Typhoon\Type;
+
+/** @psalm-check-type-exact $_emptyShape = array */
+$_emptyShape = extractType(new ArrayShapeType());
+
+/** @var ArrayShapeType<array{a?: string, 10: int}> */
+$_shapeType = new ArrayShapeType([
```

Comment:
Does this test really work?
This type cannot be inferred without a plugin.

Reply:
It doesn't, that's why there is a `@var` forced type above :)